sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CollectionExpressionsSuite.scala

```diff
@@ -2821,4 +2821,88 @@ class CollectionExpressionsSuite
       a, Literal(5), Literal.create("q", StringType)), Seq("b", "a", "c", null, "q")
     )
   }
+
+  test("Array operations on arrays of collated strings") {
```

Comment:
IMO, golden file tests already cover most of these cases.
Secondly, general agreement is to keep collation tests grouping in either CollationSuite or CollationExpressionSuite. Let's keep it that way.

Besides that LGTM.

Reply:
+1